app/src/main/java/net/gsantner/markor/format/todotxt/TodoTxtFilter.java

```diff
@@ -197,4 +216,143 @@ public static List<Group> loadSavedFilters(final Context context) {
         }
         return Collections.emptyList();
     }
+
+    // Query matching
+    // -------------------------------------------------------------------------------------------
+
+    public static boolean isMatchQuery(final TodoTxtTask task, final CharSequence query) {
```

Comment:
may I suggest to move the query matching stuff together to an own **inner** class  there?

Reply:
Hmmm, what would be benefit of this? IMO they are all nicely contained in this TodoTxtFilter class (the entire file is query matching, saving, and related things).

Reply:
my feeling just tells me that the matching stuff might might be a whole own bigger thing at some point

Reply:
:). For what it's worth, this matching query class is pretty small. 300 lines for the whole parser / matcher etc etc.

How would you separate the inner class? Right now, TodoTxtFilter contains:

1. Routines for making a query from the various filter dialogs
2. Routines for saving and loading queries
3. Routines for parsing a query

How would you suggest these be split?

IMO there is no clean split and and they are best left as is - a static class with related static functions.